test/e2e/network_test.go

```diff
@@ -782,12 +782,12 @@ func (tc *testContext) createWindowsServerJob(name, pwshCommand string, affinity
 	windowsOS := &v1.PodOS{Name: v1.Windows}
 	windowsServerImage := tc.getWindowsServerContainerImage()
 	command := []string{powerShellExe, "-command", pwshCommand}
-	return tc.createJob(name, windowsServerImage, command, &rcName, affinity, windowsOS)
+	return tc.createJob(name, windowsServerImage, command, &rcName, affinity, windowsOS, false)
 }
 
 // createJob creates a job on the cluster using the given parameters
 func (tc *testContext) createJob(name, image string, command []string, runtimeClassName *string, affinity *v1.Affinity,
-	os *v1.PodOS) (*batchv1.Job, error) {
+	os *v1.PodOS, pullSecret bool) (*batchv1.Job, error) {
```

Comment:
Rather than this being a bool, you should pass in the pull secret array.
This makes things more generic as the caller can pass in the pull-secret that should be used (or nil), and removes that logic from within the createJob function.